exploratory-notebooks/modules/fancy_time_series.py

```diff
@@ -58,52 +58,59 @@ def get_4_weeks_ago(y, m, d):
     preds = []
     y_train_dvs = y_train.select_dtypes(exclude=['object']).columns
     for index, bucket in buckets.iterrows():
-        comparison_fullyear = all_all[
-            (all_all['DECIMAL_DATE'] >= get_52_weeks_ago(
-                bucket['COMPLAINT_YEAR'],
-                bucket['COMPLAINT_MONTH'],
-                bucket['COMPLAINT_DAY'])) &
-            (all_all['DECIMAL_DATE'] < get_one_decimal_date(
-                bucket['COMPLAINT_YEAR'],
-                bucket['COMPLAINT_MONTH'],
-                bucket['COMPLAINT_DAY']))
-        ]
-        total_felonies_last_year = np.sum(
-            comparison_fullyear[y_train_dvs].values
-        )
-        comparison_lastmonth = all_all[
-            (all_all['DECIMAL_DATE'] >= get_4_weeks_ago(
-                bucket['COMPLAINT_YEAR'],
-                bucket['COMPLAINT_MONTH'],
-                bucket['COMPLAINT_DAY'])) &
-            (all_all['DECIMAL_DATE'] < get_one_decimal_date(
-                bucket['COMPLAINT_YEAR'],
-                bucket['COMPLAINT_MONTH'],
-                bucket['COMPLAINT_DAY']))
-        ]
-        total_felonies_last_month = np.sum(
-            comparison_lastmonth[y_train_dvs].values
-        )
-
-        comparison_fullyear_bucketed = comparison_fullyear.groupby([
-            'COMPLAINT_DAYOFWEEK', 'COMPLAINT_HOURGROUP', 'ADDR_PCT_CD'
-        ])[y_train_dvs].sum()
-        fullweek_pred = (
-            comparison_fullyear_bucketed *
-            (total_felonies_last_month / 4) /
-            total_felonies_last_year
-        ).reset_index()
-        pred = fullweek_pred[
-            fullweek_pred['COMPLAINT_DAYOFWEEK'] ==
-            bucket['COMPLAINT_DAYOFWEEK']
-        ].copy()
-        for fld in [
-            'COMPLAINT_YEAR',
-            'COMPLAINT_MONTH',
-            'COMPLAINT_DAY',
-        ]:
-            pred[fld] = bucket[fld]
-        preds.append(pred)
+        try:
+            comparison_fullyear = all_all[
+                (all_all['DECIMAL_DATE'] >= pd.Timestamp(get_52_weeks_ago(
```

Comment:
On line 63 you're comparing all_all['DECIMAL_DATE'] to pd.Timestamp(something else)

However, all_all['DECIMAL_DATE'] got created a few lines above (line 57), from the function get_np_dates().

get_np_dates() returns a numpy-format datetime, so you're comparing the thing on the left (not converted with pd.Timestamp) to the thing on the right (which was converted with pd.Timestamp).

Reply:
With my version of pandas, once it is stored in a pandas dataframe it is converted  to a timestamp

Reply:
I verified the type before doing this conversion